core/arch/arm/plat-k3/drivers/sa2ul_rng.c

```diff
@@ -141,6 +144,13 @@ TEE_Result sa2ul_rng_init(void)
 	io_write32(rng + RNG_FROENABLE, 0xffffff);
 
 	io_write32(rng + RNG_CONTROL, ENABLE_TRNG);
+}
+
+TEE_Result sa2ul_rng_init(void)
+{
+	rng = (vaddr_t)phys_to_virt(RNG_BASE, MEM_AREA_IO_SEC, RNG_REG_SIZE);
+
```

Comment:
Better to have check for RnG init here as well, instead of forced init 
Case of some fw already did initialization of RNG 
 @manorit2001 ?

Reply:
Wouldn't hurt ig, better to keep that check only incase TIFS does some weird things

Reply:
Done